### ISSUE TITLE
[Bug] Regression: TouchEffect NativeAmination doesn't work on images with TapGestureRecognizer

### ISSUE THREAD

**Issue by maxkoshevoi** [xamarin/XamarinCommunityToolkit#1304]
### Description

`NativeAmination` doesn't work on images. It stopped working after some update (possibly 1.1.0)

### Steps to Reproduce

1. Add native animation to any image
2. Add `TapGestureRecognizer` to the same image
3. Run the app and touch the image

(you can use sample page for testing, there is already an image there)

### Expected Behavior

Native animation works

### Actual Behavior

Nothing happens

### Basic Information

- Version with issue: 1.2.0-pre2
- Last known good version: 1.0.3 (i think)
- IDE:
- Platform Target Frameworks: 
  - iOS:  
  - Android: 11
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Workaround

Use `xct:TouchEffect.Command` instead

### Reproduction Link

There is an image at the bottom of the sample page for TouchEffect. Just add `TapGestureRecognizer` to it.

**Comment by maxkoshevoi** [comment]
CC: @AndreiMisiukevich

**Comment by maxkoshevoi** [comment]
@AndreiMisiukevich I was able to partially fix the bug. Ripple effect appears one on first touch. On any subsequent touches `OnTouch` just isn't invoked. It's invoked for `MotionEventActions.Down` and `MotionEventActions.Up` of the first touch and that's it.

**Comment by AndreiMisiukevich** [comment]
@maxkoshevoi great investingation) reviewing PR

**Comment by AndreiMisiukevich** [comment]
Hey @maxkoshevoi what android version do you use for testing?

It works fine for me on Android API 28

![image](https://user-images.githubusercontent.com/10124814/118413704-f6688680-b6a0-11eb-9b95-e7523fed576f.png)

**Comment by AndreiMisiukevich** [comment]
@maxkoshevoi unfortunately with your current fix another bug starts to reproduce for me :(
This one https://github.com/xamarin/XamarinCommunityToolkit/pull/817

That's why I didn't set the entire View to rippleView

I presume you test on Android 5, no?

**Comment by AndreiMisiukevich** [comment]
Tried Android 11 (works fine w/out proposed changes as well).

Could you please re-test?

**Comment by maxkoshevoi** [comment]
Hm, you are correct it does work in sample app, but it doesn't in my app for some reason (during testing I mixed up something and thought that it doesn't work in general).

I'll do some more tests later to investigate what's going on in my case.

Thank you for checking the issue.

**Comment by AndreiMisiukevich** [comment]
@maxkoshevoi thank you :)
Hm, interesting. Maybe it's a linker issue... again...

**Comment by maxkoshevoi** [comment]
@AndreiMisiukevich Looks like it's `TapGestureRecognizer` that's doing it.

If I remove this
```
<Image.GestureRecognizers>
    <TapGestureRecognizer Command="{Binding OptionsCommand}" />
</Image.GestureRecognizers>
```

and add this 
```
xct:TouchEffect.Command="{Binding OptionsCommand}"
```

animation magically begins to work

**Comment by maxkoshevoi** [comment]
> Maybe it's a linker issue... again...

It's definitely not linker this time. I disabled it during my tests

**Comment by maxkoshevoi** [comment]
General question. Why does `TouchEffect` needs `Command` anyway? Are there some benefits/drawbacks of using it instead of `TapGestureRecognizer `?

**Comment by AndreiMisiukevich** [comment]
@maxkoshevoi well, actually TouchEffect should work with TapGestureRecognizer as well. But in the case of Image (on Android), XF implementation of TapGestureRecognize intercepts gestures (prevents passing touch events to image renderer).

It happens because in the case of Image we don't control ripple animation (we put it to the foreground of view instead of the Group's background like we do for Layouts).


I presume if you try to attach touch effect to StackLayout with TapGesture, it will work as expected.


Basically, we can try to work around it, but... It's better to use Command of TouchEffect because it has more button-like behavior.

Try to press the view with the touch effect, then slide your finger outside of view, and then slide the finger back to view (all actions should be without releasing your finger).
Once you release your finger the TouchEffect command will be raised. (In case of TapGestureRecognizer it won't be)

**Comment by maxkoshevoi** [comment]
So it's better to use `Command`, got it. 
Can you add a check when setting ripple to the view itself if there is tap recognizer attached to the view, and throw exception if it is (that says that native animation won't work and user needs to use `Command` instead)?

**Comment by AndreiMisiukevich** [comment]
Would be better to work around this issue somehow. Need to think.
